src/endpoints/user/delete/delete_current_user.py

```diff
@@ -25,7 +26,11 @@ async def delete_current_user(
     HTTP_STATUS_CODE_204
     """
     try:
-        db.execute(delete(User).where(User.id == user.get("id")))
+        user_to_delete = db.scalar(select(User).where(User.id == user.get("id")))
```

Comment:
Shouldn't we add another check here to see if the user is not already deleted? I see you have done that in the other endpoint but not in this one.

Reply:
That was required in the other route but not here since that route was fetching the user based on the user_id provided as path parameter( since path parameter is provided by the front end that could be anything ) How ever this route fetches the user from the JWT Token ( Since a deleted user can not log in hence can not have a JWT Token ) So having JWT token insures that user is not deleted.

Reply:
That makes sense, but this also leads to another question that Is the JWT of the user being invalidated when that user is deleted? If not then this means the user can make API calls even after being deleted.

Reply:
Damn! that is a good edge case. Adding token to the black list will work in case of `delete_current_user` where we have access to JWT token of the user being deleted, but is case of `delete_user_id` we don't have have access to the JWT token of the user being deleted. Hence that token still being valid will allow the deleted user to make requests. 
**_However_**
I was thinking what we can do is edit `get_current_user` and `get_current_librarian` to check for deleted users also, just like the way we did for the black listed tokens. In that way any logged in user that is deleted by the librarian after being logged in will not be able to make any further requests and it won't break any previous code.

Reply:
But that would mean a db call every time the jwt is being validated. Better option would be find a way to add the token of the user to the blacklist.